### PR TITLE
Make OCSP stapling tests more stable

### LINKED ISSUE #97836: System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid test failure
## Build Information
Build: https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=549625
Build error leg or test failing: System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid
Pull request: https://github.com/dotnet/runtime/pull/96931

## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#how-to-fill-out-a-known-issue-error-section).

```json
{
  "ErrorMessage": "System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid",
  "ErrorPattern": "",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=549625
**Error message validated:** `System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid`
**Result validation:** :white_check_mark: Known issue matched with the provided build.
**Validation performed at:** 2/1/2024 8:55:26 PM UTC

### Report

|Build|Definition|Test|Pull Request|
|---|---|---|---|
|[563743](https://dev.azure.com/dnceng-public/public/_build/results?buildId=563743)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=563743&view=ms.vss-test-web.build-test-results-tab&runId=13560472&resultId=156684)|dotnet/runtime#98330|
|[565864](https://dev.azure.com/dnceng-public/public/_build/results?buildId=565864)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=565864&view=ms.vss-test-web.build-test-results-tab&runId=13525926&resultId=147294)|dotnet/runtime#98352|
|[565668](https://dev.azure.com/dnceng-public/public/_build/results?buildId=565668)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid](https://dev.azure.com/dnceng-public/public/_build/results?buildId=565668&view=ms.vss-test-web.build-test-results-tab&runId=13521608&resultId=157883)|dotnet/runtime#98400|
|[565641](https://dev.azure.com/dnceng-public/public/_build/results?buildId=565641)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=565641&view=ms.vss-test-web.build-test-results-tab&runId=13519370&resultId=138099)|dotnet/runtime#97079|
|[565028](https://dev.azure.com/dnceng-public/public/_build/results?buildId=565028)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=565028&view=ms.vss-test-web.build-test-results-tab&runId=13500678&resultId=150410)|dotnet/runtime#98380|
|[564516](https://dev.azure.com/dnceng-public/public/_build/results?buildId=564516)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=564516&view=ms.vss-test-web.build-test-results-tab&runId=13487236&resultId=160234)|dotnet/runtime#98337|
|[564485](https://dev.azure.com/dnceng-public/public/_build/results?buildId=564485)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid](https://dev.azure.com/dnceng-public/public/_build/results?buildId=564485&view=ms.vss-test-web.build-test-results-tab&runId=13486166&resultId=152177)|dotnet/runtime#98351|
|[564408](https://dev.azure.com/dnceng-public/public/_build/results?buildId=564408)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid](https://dev.azure.com/dnceng-public/public/_build/results?buildId=564408&view=ms.vss-test-web.build-test-results-tab&runId=13484712&resultId=151753)|dotnet/runtime#97388|
|[564348](https://dev.azure.com/dnceng-public/public/_build/results?buildId=564348)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid](https://dev.azure.com/dnceng-public/public/_build/results?buildId=564348&view=ms.vss-test-web.build-test-results-tab&runId=13483690&resultId=156962)|dotnet/runtime#97508|
|[564288](https://dev.azure.com/dnceng-public/public/_build/results?buildId=564288)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid](https://dev.azure.com/dnceng-public/public/_build/results?buildId=564288&view=ms.vss-test-web.build-test-results-tab&runId=13481406&resultId=148997)|dotnet/runtime#98350|
|[2377467](https://dev.azure.com/dnceng/internal/_build/results?buildId=2377467)|dotnet-runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng//internal/_build/results?buildId=2377467&view=ms.vss-test-web.build-test-results-tab&runId=53063251&resultId=165652)||
|[564172](https://dev.azure.com/dnceng-public/public/_build/results?buildId=564172)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid](https://dev.azure.com/dnceng-public/public/_build/results?buildId=564172&view=ms.vss-test-web.build-test-results-tab&runId=13477378&resultId=156296)|dotnet/runtime#98340|
|[564095](https://dev.azure.com/dnceng-public/public/_build/results?buildId=564095)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid](https://dev.azure.com/dnceng-public/public/_build/results?buildId=564095&view=ms.vss-test-web.build-test-results-tab&runId=13476086&resultId=148965)||
|[564061](https://dev.azure.com/dnceng-public/public/_build/results?buildId=564061)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=564061&view=ms.vss-test-web.build-test-results-tab&runId=13474950&resultId=140721)||
|[564029](https://dev.azure.com/dnceng-public/public/_build/results?buildId=564029)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=564029&view=ms.vss-test-web.build-test-results-tab&runId=13474080&resultId=149673)|dotnet/runtime#97079|
|[563865](https://dev.azure.com/dnceng-public/public/_build/results?buildId=563865)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=563865&view=ms.vss-test-web.build-test-results-tab&runId=13469902&resultId=151852)|dotnet/runtime#98129|
|[563637](https://dev.azure.com/dnceng-public/public/_build/results?buildId=563637)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid](https://dev.azure.com/dnceng-public/public/_build/results?buildId=563637&view=ms.vss-test-web.build-test-results-tab&runId=13463444&resultId=138010)|dotnet/runtime#98320|
|[563346](https://dev.azure.com/dnceng-public/public/_build/results?buildId=563346)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=563346&view=ms.vss-test-web.build-test-results-tab&runId=13456948&resultId=152502)|dotnet/runtime#98318|
|[563037](https://dev.azure.com/dnceng-public/public/_build/results?buildId=563037)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=563037&view=ms.vss-test-web.build-test-results-tab&runId=13448140&resultId=149032)|dotnet/runtime#98310|
|[562830](https://dev.azure.com/dnceng-public/public/_build/results?buildId=562830)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=562830&view=ms.vss-test-web.build-test-results-tab&runId=13441696&resultId=144822)|dotnet/runtime#98301|
|[562788](https://dev.azure.com/dnceng-public/public/_build/results?buildId=562788)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=562788&view=ms.vss-test-web.build-test-results-tab&runId=13440510&resultId=150830)|dotnet/runtime#98199|
|[562540](https://dev.azure.com/dnceng-public/public/_build/results?buildId=562540)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=562540&view=ms.vss-test-web.build-test-results-tab&runId=13435464&resultId=157037)|dotnet/runtime#97508|
|[562353](https://dev.azure.com/dnceng-public/public/_build/results?buildId=562353)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=562353&view=ms.vss-test-web.build-test-results-tab&runId=13429574&resultId=148651)|dotnet/runtime#98289|
|[562249](https://dev.azure.com/dnceng-public/public/_build/results?buildId=562249)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=562249&view=ms.vss-test-web.build-test-results-tab&runId=13425764&resultId=142098)||
|[562194](https://dev.azure.com/dnceng-public/public/_build/results?buildId=562194)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=562194&view=ms.vss-test-web.build-test-results-tab&runId=13424310&resultId=149059)|dotnet/runtime#98273|
|[562110](https://dev.azure.com/dnceng-public/public/_build/results?buildId=562110)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_AfterExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=562110&view=ms.vss-test-web.build-test-results-tab&runId=13421050&resultId=150025)|dotnet/runtime#98284|
|[562101](https://dev.azure.com/dnceng-public/public/_build/results?buildId=562101)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=562101&view=ms.vss-test-web.build-test-results-tab&runId=13419738&resultId=160425)|dotnet/runtime#98273|
|[562000](https://dev.azure.com/dnceng-public/public/_build/results?buildId=562000)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid](https://dev.azure.com/dnceng-public/public/_build/results?buildId=562000&view=ms.vss-test-web.build-test-results-tab&runId=13416438&resultId=145849)|dotnet/runtime#98279|
|[561955](https://dev.azure.com/dnceng-public/public/_build/results?buildId=561955)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=561955&view=ms.vss-test-web.build-test-results-tab&runId=13416158&resultId=149476)||
|[561889](https://dev.azure.com/dnceng-public/public/_build/results?buildId=561889)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=561889&view=ms.vss-test-web.build-test-results-tab&runId=13414240&resultId=149442)|dotnet/runtime#98277|
|[561792](https://dev.azure.com/dnceng-public/public/_build/results?buildId=561792)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_AfterExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=561792&view=ms.vss-test-web.build-test-results-tab&runId=13410858&resultId=164006)|dotnet/runtime#98187|
|[561770](https://dev.azure.com/dnceng-public/public/_build/results?buildId=561770)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=561770&view=ms.vss-test-web.build-test-results-tab&runId=13410158&resultId=156467)|dotnet/runtime#98273|
|[561619](https://dev.azure.com/dnceng-public/public/_build/results?buildId=561619)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=561619&view=ms.vss-test-web.build-test-results-tab&runId=13405154&resultId=145148)|dotnet/runtime#98238|
|[561602](https://dev.azure.com/dnceng-public/public/_build/results?buildId=561602)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid](https://dev.azure.com/dnceng-public/public/_build/results?buildId=561602&view=ms.vss-test-web.build-test-results-tab&runId=13403820&resultId=149502)||
|[561530](https://dev.azure.com/dnceng-public/public/_build/results?buildId=561530)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=561530&view=ms.vss-test-web.build-test-results-tab&runId=13402422&resultId=153097)|dotnet/runtime#97508|
|[561405](https://dev.azure.com/dnceng-public/public/_build/results?buildId=561405)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid](https://dev.azure.com/dnceng-public/public/_build/results?buildId=561405&view=ms.vss-test-web.build-test-results-tab&runId=13396876&resultId=159227)|dotnet/runtime#98233|
|[561377](https://dev.azure.com/dnceng-public/public/_build/results?buildId=561377)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=561377&view=ms.vss-test-web.build-test-results-tab&runId=13396110&resultId=153198)|dotnet/runtime#98007|
|[561096](https://dev.azure.com/dnceng-public/public/_build/results?buildId=561096)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=561096&view=ms.vss-test-web.build-test-results-tab&runId=13389654&resultId=153459)|dotnet/runtime#98048|
|[560936](https://dev.azure.com/dnceng-public/public/_build/results?buildId=560936)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=560936&view=ms.vss-test-web.build-test-results-tab&runId=13384042&resultId=156493)|dotnet/runtime#98238|
|[560873](https://dev.azure.com/dnceng-public/public/_build/results?buildId=560873)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=560873&view=ms.vss-test-web.build-test-results-tab&runId=13382836&resultId=148151)|dotnet/runtime#98243|
|[560719](https://dev.azure.com/dnceng-public/public/_build/results?buildId=560719)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid](https://dev.azure.com/dnceng-public/public/_build/results?buildId=560719&view=ms.vss-test-web.build-test-results-tab&runId=13377998&resultId=160925)||
|[560679](https://dev.azure.com/dnceng-public/public/_build/results?buildId=560679)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=560679&view=ms.vss-test-web.build-test-results-tab&runId=13377712&resultId=154697)|dotnet/runtime#98199|
|[559193](https://dev.azure.com/dnceng-public/public/_build/results?buildId=559193)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=559193&view=ms.vss-test-web.build-test-results-tab&runId=13375586&resultId=152427)|dotnet/runtime#98186|
|[560605](https://dev.azure.com/dnceng-public/public/_build/results?buildId=560605)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid](https://dev.azure.com/dnceng-public/public/_build/results?buildId=560605&view=ms.vss-test-web.build-test-results-tab&runId=13374538&resultId=158951)|dotnet/runtime#98238|
|[560434](https://dev.azure.com/dnceng-public/public/_build/results?buildId=560434)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=560434&view=ms.vss-test-web.build-test-results-tab&runId=13371354&resultId=158756)|dotnet/runtime#98199|
|[560023](https://dev.azure.com/dnceng-public/public/_build/results?buildId=560023)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=560023&view=ms.vss-test-web.build-test-results-tab&runId=13357810&resultId=158125)|dotnet/runtime#98218|
|[559896](https://dev.azure.com/dnceng-public/public/_build/results?buildId=559896)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=559896&view=ms.vss-test-web.build-test-results-tab&runId=13354900&resultId=150933)|dotnet/runtime#97640|
|[559736](https://dev.azure.com/dnceng-public/public/_build/results?buildId=559736)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=559736&view=ms.vss-test-web.build-test-results-tab&runId=13351076&resultId=159611)||
|[559667](https://dev.azure.com/dnceng-public/public/_build/results?buildId=559667)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=559667&view=ms.vss-test-web.build-test-results-tab&runId=13348474&resultId=154308)|dotnet/runtime#98198|
|[559390](https://dev.azure.com/dnceng-public/public/_build/results?buildId=559390)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid](https://dev.azure.com/dnceng-public/public/_build/results?buildId=559390&view=ms.vss-test-web.build-test-results-tab&runId=13341028&resultId=155950)|dotnet/runtime#98187|
|[559177](https://dev.azure.com/dnceng-public/public/_build/results?buildId=559177)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_AfterExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=559177&view=ms.vss-test-web.build-test-results-tab&runId=13334572&resultId=157607)|dotnet/runtime#98129|
|[559023](https://dev.azure.com/dnceng-public/public/_build/results?buildId=559023)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=559023&view=ms.vss-test-web.build-test-results-tab&runId=13329176&resultId=154767)|dotnet/runtime#97350|
|[559039](https://dev.azure.com/dnceng-public/public/_build/results?buildId=559039)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=559039&view=ms.vss-test-web.build-test-results-tab&runId=13328200&resultId=151985)||
|[558575](https://dev.azure.com/dnceng-public/public/_build/results?buildId=558575)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid](https://dev.azure.com/dnceng-public/public/_build/results?buildId=558575&view=ms.vss-test-web.build-test-results-tab&runId=13317646&resultId=147460)|dotnet/runtime#97508|
|[558239](https://dev.azure.com/dnceng-public/public/_build/results?buildId=558239)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid](https://dev.azure.com/dnceng-public/public/_build/results?buildId=558239&view=ms.vss-test-web.build-test-results-tab&runId=13308086&resultId=158750)|dotnet/runtime#98133|
|[558060](https://dev.azure.com/dnceng-public/public/_build/results?buildId=558060)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid](https://dev.azure.com/dnceng-public/public/_build/results?buildId=558060&view=ms.vss-test-web.build-test-results-tab&runId=13303146&resultId=157269)|dotnet/runtime#97356|
|[557848](https://dev.azure.com/dnceng-public/public/_build/results?buildId=557848)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=557848&view=ms.vss-test-web.build-test-results-tab&runId=13296578&resultId=162263)|dotnet/runtime#98127|
|[557541](https://dev.azure.com/dnceng-public/public/_build/results?buildId=557541)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=557541&view=ms.vss-test-web.build-test-results-tab&runId=13289120&resultId=153064)|dotnet/runtime#97935|
|[557349](https://dev.azure.com/dnceng-public/public/_build/results?buildId=557349)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_AfterExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=557349&view=ms.vss-test-web.build-test-results-tab&runId=13282608&resultId=141504)|dotnet/runtime#98096|
|[557374](https://dev.azure.com/dnceng-public/public/_build/results?buildId=557374)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid](https://dev.azure.com/dnceng-public/public/_build/results?buildId=557374&view=ms.vss-test-web.build-test-results-tab&runId=13283156&resultId=155062)|dotnet/runtime#98048|
|[557254](https://dev.azure.com/dnceng-public/public/_build/results?buildId=557254)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=557254&view=ms.vss-test-web.build-test-results-tab&runId=13280360&resultId=153570)|dotnet/runtime#98094|
|[557214](https://dev.azure.com/dnceng-public/public/_build/results?buildId=557214)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=557214&view=ms.vss-test-web.build-test-results-tab&runId=13279924&resultId=157585)|dotnet/runtime#97814|
|[557143](https://dev.azure.com/dnceng-public/public/_build/results?buildId=557143)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_BeforeExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=557143&view=ms.vss-test-web.build-test-results-tab&runId=13277760&resultId=153378)|dotnet/runtime#98104|
|[556841](https://dev.azure.com/dnceng-public/public/_build/results?buildId=556841)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.RefreshOcspResponse_AfterExpiration](https://dev.azure.com/dnceng-public/public/_build/results?buildId=556841&view=ms.vss-test-web.build-test-results-tab&runId=13270508&resultId=157622)|dotnet/runtime#98083|
|[556818](https://dev.azure.com/dnceng-public/public/_build/results?buildId=556818)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid](https://dev.azure.com/dnceng-public/public/_build/results?buildId=556818&view=ms.vss-test-web.build-test-results-tab&runId=13269774&resultId=152422)|dotnet/runtime#98085|
|[556607](https://dev.azure.com/dnceng-public/public/_build/results?buildId=556607)|dotnet/runtime|[System.Net.Security.Tests.SslStreamCertificateContextOcspLinuxTests.FetchOcspResponse_FirstInvalidThenValid](https://dev.azure.com/dnceng-public/public/_build/results?buildId=556607&view=ms.vss-test-web.build-test-results-tab&runId=13263018&resultId=152752)||
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|57|

### PR DESCRIPTION
**Pull request by rzikm**
Fixes #97836, #97779 and #97936.

The instability of the tests was caused by the tests racing with refreshing the OCSP staple on the background. This PR adds waiting for the background refresh to complete where it was assumed it will complete fast enough.

I kept the tests running for a bit on my machine, and did not reproduce the failure after the changes.

### CODE REVIEW

File: src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.Linux.cs [hunk 1]
```diff
@@ -154,6 +154,17 @@ private SslStreamCertificateContext(X509Certificate2 target, ReadOnlyCollection<
             return ValueTask.FromResult(_ocspResponse);
         }
 
+        internal ValueTask<byte[]?> WaitForPendingOcspFetchAsync()
```
[the remark this answers is not in the document]
Comment (vcsjones) [reply src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.Linux.cs:157]: The unit tests pull in the source files directly. https://github.com/dotnet/runtime/blob/2d5c396eaf795d0c31f934b81b396509a9c5fec5/src/libraries/System.Net.Security/tests/UnitTests/System.Net.Security.Unit.Tests.csproj#L54

[the remark this answers is not in the document]
Comment (wfurt) [reply src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.Linux.cs:157]: ahh. these are UniteTests not FunctionalTests...

**Review by wfurt — APPROVED** [review]
LGTM

File: src/libraries/System.Net.Security/tests/UnitTests/SslStreamCertificateContextOcspLinuxTests.cs [hunk 2]
```diff
@@ -18,6 +18,7 @@ namespace System.Net.Security.Tests;
 public class SslStreamCertificateContextOcspLinuxTests
 {
     [Fact]
+    [PlatformSpecific(TestPlatforms.Linux)]
```
Comment (bartonjs) [inline src/libraries/System.Net.Security/tests/UnitTests/SslStreamCertificateContextOcspLinuxTests.cs:21]: The class is named "OcspLinuxTests", can/should this attribute just be put on the class? (I think it works at that level)

**Comment by rzikm** [comment]
CI Failures are unrelated.

**Pull request merged**